api/v1/cmd/example-executor/main.go

```diff
@@ -104,43 +111,35 @@ func run(cfg config.Config) {
 			log.Printf("failed to re-establish subscription with agent within %v, aborting", cfg.RecoveryTimeout)
 			return
 		}
+		log.Println("waiting for reconnect timeout")
 		<-shouldReconnect // wait for some amount of time before retrying subscription
 	}
 }
 
-// unacknowledgedTasks is a functional option that sets the value of the UnacknowledgedTasks
-// field of a Subscribe call.
-func unacknowledgedTasks(state *internalState) executor.CallOpt {
-	return func(call *executor.Call) {
-		if n := len(state.unackedTasks); n > 0 {
-			unackedTasks := make([]mesos.TaskInfo, 0, n)
-			for k := range state.unackedTasks {
-				unackedTasks = append(unackedTasks, state.unackedTasks[k])
-			}
-			call.Subscribe.UnacknowledgedTasks = unackedTasks
-		} else {
-			call.Subscribe.UnacknowledgedTasks = nil
+// unacknowledgedTasks generates the value of the UnacknowledgedTasks field of a Subscribe call.
+func unacknowledgedTasks(state *internalState) (result []mesos.TaskInfo) {
```

Comment:
I think `unacknowledgedTasks` should be a function on `*internalState` but it's a minor issue.

Reply:
I could go either way. Maybe tackle in a follow-up PR